projtools/tests/test_data.py

```diff
@@ -3,34 +3,73 @@
 
 def test_change_date_cols():
     """Test that the date columns are changed correctly"""
     change_date_cols = data._Data__change_date_cols
-    assert change_date_cols(pd.Index(["201001", "202108"])) == {"201001": "2010-01", "202108": "2021-08"}
+    df = pd.DataFrame({"201001": [1,2,3], "202108": [1,2,3]})
+    assert change_date_cols(df, 0).equals(pd.DataFrame({"2010-01": [1,2,3], "2021-08": [1,2,3]}))
+    assert change_date_cols(df, 1).equals(pd.DataFrame({"201001": [1,2,3], "2021-08": [1,2,3]}))
+    assert change_date_cols(df, 2).equals(pd.DataFrame({"201001": [1,2,3], "202108": [1,2,3]}))
+
+def test_make_multi_index():
+    """Test that the multi-index is made correctly"""
+    make_multi_index = data._Data__make_multi_index
+    df = pd.DataFrame({"2010-01": [1,2,3], "2021-08": [1,2,3], "2021-09": [1,2,3], "2021-10": [1,2,3], "2021-11": [1,2,3], "2021-12": [1,2,3]})
+    assert list(make_multi_index(df, 1).columns) == [('2021', '08'),('2021', '09'),('2021', '10'),('2021', '11'),('2021', '12')]
+    assert list(make_multi_index(df, 1).index) == [1,2,3]
+    assert list(make_multi_index(df, 3).columns) == [('2021', '10'),('2021', '11'),('2021', '12')]
+    assert list(make_multi_index(df, 3).index) == [(1, 1, 1), (2, 2, 2), (3, 3, 3)]
+
```

Comment:
what is the logic of these tests?

Reply:
it basically checks that based on the input dataframe, the right columns and rows are created